chia/server/server.py

```diff
@@ -99,6 +99,13 @@ def ssl_context_for_client(
     return ssl_context
 
 
+def calculate_node_id(cert_path: Path) -> bytes32:
+    pem_cert = x509.load_pem_x509_certificate(cert_path.read_bytes(), default_backend())
```

Comment:
I think this loads `cert_path` a second time, though I also think that the other load is buried down in Python stdlib code so we may not be able to save the double load.  Unless maybe we already have the `pem_cert` object from the previous load and could pass it in?  I'm not sure what all is in the `ssl_context`.